Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAssetHandler.cpp

```diff
@@ -36,14 +36,155 @@ namespace AZ
+                                pendingAssetInfo.m_mipChainAssetSubIds.push_back(mipChainAssetId.m_subId);
+                                Data::AssetBus::MultiHandler::BusConnect(mipChainAssetId);
+                            }
+                        }
+
+                        if (pendingAssetInfo.m_mipChainAssetSubIds.size() > 0)
+                        {
+                            AZStd::scoped_lock<AZStd::mutex> lock(m_accessPendingAssetsMutex);
+                            m_pendingReloadImageAsset[imageGuid] = AZStd::move(pendingAssetInfo);
+                        }
+                    }
                 }
             }
 
             return loadResult;
         }
 
+        void StreamingImageAssetHandler::OnAssetReady(Data::Asset<Data::AssetData> asset)
+        {
+            HandleMipChainAssetLoad(asset, true);
+        }
+
+        void StreamingImageAssetHandler::OnAssetError(AZ::Data::Asset<AZ::Data::AssetData> asset)
+        {
+            HandleMipChainAssetLoad(asset, false);
+        }
+        
+        void StreamingImageAssetHandler::OnAssetReloaded(Data::Asset<Data::AssetData> asset)
+        {
+            HandleMipChainAssetLoad(asset, true);
+        }
+
+        void StreamingImageAssetHandler::OnAssetReloadError(Data::Asset<Data::AssetData> asset)
+        {
+            HandleMipChainAssetLoad(asset, false);
+        }
+        
+        void StreamingImageAssetHandler::HandleMipChainAssetBuses(Data::Asset<Data::AssetData> streamingImageAsset, bool connect)
+        {
+            StreamingImageAsset* assetData = streamingImageAsset.GetAs<StreamingImageAsset>();
+            for (auto& mipChainAsset : assetData->m_mipChains)
+            {
+                // Need to skip tail mipchain which is in the list and doesn't have an asset id.
+                if (mipChainAsset.m_asset.GetId().IsValid())
+                {
+                    if (connect)
+                    {
+                        Data::AssetBus::MultiHandler::BusConnect(mipChainAsset.m_asset.GetId());
+                    }
+                    else
+                    {
+                        Data::AssetBus::MultiHandler::BusDisconnect(mipChainAsset.m_asset.GetId());
+                    }
+                }
+            }
+        }
+
+        void StreamingImageAssetHandler::HandleMipChainAssetLoad(Data::Asset<Data::AssetData> asset, bool isLoadSuccess)
+        {
+            AZStd::scoped_lock<AZStd::mutex> lock(m_accessPendingAssetsMutex);
+            auto itr = m_pendingReloadImageAsset.find(asset.GetId().m_guid);
+            if (itr == m_pendingReloadImageAsset.end())
+            {
+                return;
+            }
+
+            if (!isLoadSuccess)
+            {
+                HandleMipChainAssetBuses(itr->second.m_imageAsset, false);
+                Data::AssetManagerBus::Broadcast(&Data::AssetManagerBus::Events::OnAssetReloadError, itr->second.m_imageAsset);
+                m_pendingReloadImageAsset.erase(itr);
```

Comment:
Suggestion: You might want to flip the order of these lines and release the lock before the call, since you never know just how much logic is going to execute during the OnAssetReloadError.